references/classification/train_quantization.py

```diff
@@ -0,0 +1,254 @@
+from __future__ import print_function
+import datetime
+import os
+import time
+import sys
+import copy
+
+import torch
+import torch.utils.data
+from torch import nn
+import torchvision
+import torch.quantization
+import utils
+from train import train_one_epoch, evaluate, load_data
+
+
+def main(args):
+
+    if args.output_dir:
+        utils.mkdir(args.output_dir)
+
+    utils.init_distributed_mode(args)
+    print(args)
+
+    if args.post_training_quantize and args.distributed:
+        raise RuntimeError("Post training quantization example should not be performed "
+                           "on distributed mode")
+
+    # Set backend engine to ensure that quantized model runs on the correct kernels
+    if args.backend not in torch.backends.quantized.supported_engines:
+        raise RuntimeError("Quantized backend not supported: " + str(args.backend))
+    torch.backends.quantized.engine = args.backend
+
+    device = torch.device(args.device)
+    torch.backends.cudnn.benchmark = True
+
+    # Data loading code
+    print("Loading data")
+    train_dir = os.path.join(args.data_path, 'train')
+    val_dir = os.path.join(args.data_path, 'val')
+
+    dataset, dataset_test, train_sampler, test_sampler = load_data(train_dir, val_dir,
+                                                                   args.cache_dataset, args.distributed)
+    data_loader = torch.utils.data.DataLoader(
+        dataset, batch_size=args.batch_size,
+        sampler=train_sampler, num_workers=args.workers, pin_memory=True)
+
+    data_loader_test = torch.utils.data.DataLoader(
+        dataset_test, batch_size=args.eval_batch_size,
+        sampler=test_sampler, num_workers=args.workers, pin_memory=True)
+
+    print("Creating model", args.model)
+    # when training quantized models, we always start from a pre-trained fp32 reference model
+    model = torchvision.models.quantization.__dict__[args.model](pretrained=True, quantize=args.test_only)
+    model.to(device)
+
+    if not (args.test_only or args.post_training_quantize):
+        model.fuse_model()
```

Comment:
@fmassa : Thanks for all the improvements!. Wondering why we are removing this part for quantization aware training?
``` python 
    if args.distributed and args.sync_bn:
        model = torch.nn.SyncBatchNorm.convert_sync_batchnorm(model)	        model = torch.nn.SyncBatchNorm.convert_sync_batchnorm(model)
```

Reply:
When I tried `SyncBatchNorm` for classification it didn't improve results (and actually made results a bit worse).
If users use `SyncBatchNorm` in the quantization setting, I'm afraid it won't get fused afterwards by the quantization fusion code, which makes strong assumptions [on the type of the module being `BatchNorm`](https://github.com/pytorch/pytorch/blob/1a5d32d894fe50e72de167d1ff7270018145b01f/torch/quantization/fuse_modules.py#L88-L89), and not `SyncBatchNorm`.
Given those points, I think it's better to remove support for `SyncBatchNorm` for now.